src/btree/bt_sync_obsolete.c

```diff
@@ -559,9 +597,21 @@ __checkpoint_cleanup_eligibility(WT_SESSION_IMPL *session, const char *uri, cons
     if (newest_stop_durable_ts != WT_TS_NONE)
         return (true);
 
+    /*
+     * Upon restart, the write generation number will be lower than the connection one. For this
+     * reason, don't rely on the transaction as the transaction will be very likely bigger than the
+     * current oldest one.
+     */
+    if (write_gen < S2C(session)->base_write_gen)
```

Comment:
Why we need to do this here?

Reply:
Let me know if 1b319ef172ba19326a4a105297e453493dd9c1ee makes more sense